### PR TITLE
Added Prime Number in Odin

### PR DESCRIPTION
**Pull request by nallovint**
Congrats on taking the first step to contributing to the Sample Programs repository maintained by [The Renegade Coder][renegade-coder]! 
For simplicity, please make sure that your pull request includes one and only one contribution.

Please fill _one_ of the sections below as applicable.
Please also add any other relevant information to the Notes section at the bottom.
You may delete or just ignore any other sections.
For more information please refer to our [contributing documentation][contributing]

## I Am Adding a New Code Snippet in an Existing Language

- [x] I named the pull request using `Add {PROJECT} in {LANGUAGE}` format
  
  
## Other Notes

[renegade-coder]: https://therenegadecoder.com/
[contributing-plagiarism]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#plagiarism
[contributing-new-project]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#requirements-for-a-new-project
[contributing-new-language]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#requirements-for-a-new-language
[contributing-readme]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#create-readmes
[contributing-tests-in-detail]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#tests-in-detail
[contributing]: ../CONTRIBUTING.md
[sample-programs-project-list]: https://sampleprograms.io/projects/
[contributing-modifications]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#modifying-existing-code-snippets

### CODE REVIEW

File: archive/o/odin/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+prime-number
```
Comment (jrg94) [inline archive/o/odin/.gitignore:1]: I would delete this file, so it's not in the repo itself.

File: archive/o/odin/prime-number.odin [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		return
+	}
+	
+	if is_prime(n) {
+		fmt.println("prime")
+	} else {
+		fmt.println("composite")
+	}
+}
+// uncomment to run file locally
```
Comment (jrg94) [inline archive/o/odin/prime-number.odin:49]: I think you'll need to expose this code section for our testing framework. Though, I suppose we'll wait and see what happens with testing.

File: ols.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+    "$schema": "https://raw.githubusercontent.com/DanielGavin/ols/master/misc/ols.schema.json",
+    "enable_document_symbols": true,
+    "enable_hover": true,
+    "enable_snippets": true
+}
```
Comment (jrg94) [inline ols.json:6]: I would delete this as well.

File: archive/o/odin/reverse-string.odin [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+reverse_string :: proc(str : string) {
+    sb := strings.builder_make(0, len(str))
+    #reverse for ch in str {
+        strings.write_rune(&sb, ch)
+    }
+    fmt.println(strings.to_string(sb))
+}
+// uncomment below if you want to run locally
+// main :: proc() {
```
Comment (jrg94) [inline archive/o/odin/reverse-string.odin:15]: See comment above.

**Review by jrg94 — CHANGES_REQUESTED** [review]
Everything looks good so far! I think there are some minor changes needed. Maybe with future solutions you can put them on separate branches.

File: archive/o/odin/reverse-string.odin (hunk 4 above)
Comment (nallovint) [reply archive/o/odin/reverse-string.odin]: Done

**Comment by jrg94** [comment]
Okay, tests failed! I don't know basically anything about this language, but it looks like the Docker image is expected a main function for testing:

```
=========================== short test summary info ============================
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input 0] - assert "invalid flag...cedure 'main'" == 'composite'
  - composite
  + invalid flag: 0
  + /src/prime-number.odin(1:2) undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input large prime] - assert "invalid flag...cedure 'main'" == 'prime'
  - prime
  + invalid flag: 3727
  + /src/prime-number.odin(1:2) undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input 1] - assert "invalid flag...cedure 'main'" == 'composite'
  - composite
  + invalid flag: 1
  + /src/prime-number.odin(1:2) undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_invalid[odin/prime-number.odin-no input] - assert "/src/prime-n...cedure 'main'" == 'Usage: pleas...ative integer'
  - Usage: please input a non-negative integer
  + /src/prime-number.odin(1:2) Undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input 2] - assert "invalid flag...cedure 'main'" == 'prime'
  - prime
  + invalid flag: 2
  + /src/prime-number.odin(1:2) undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_invalid[odin/prime-number.odin-empty input] - assert "/src/prime-n...cedure 'main'" == 'Usage: pleas...ative integer'
  - Usage: please input a non-negative integer
  + /src/prime-number.odin(1:2) Undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input small composite] - assert "invalid flag...cedure 'main'" == 'composite'
  - composite
  + invalid flag: 4
  + /src/prime-number.odin(1:2) undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_invalid[odin/prime-number.odin-invalid input: not a number] - assert "Invalid flag...cedure 'main'" == 'Usage: pleas...ative integer'
  - Usage: please input a non-negative integer
  + Invalid flag: a
  + /src/prime-number.odin(1:2) Undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input small prime] - assert "invalid flag...cedure 'main'" == 'prime'
  - prime
  + invalid flag: 7
  + /src/prime-number.odin(1:2) undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_invalid[odin/prime-number.odin-invalid input: not an integer] - assert "Invalid flag...cedure 'main'" == 'Usage: pleas...ative integer'
  - Usage: please input a non-negative integer
  + Invalid flag: 6.7
  + /src/prime-number.odin(1:2) Undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_invalid[odin/prime-number.odin-invalid input: negative] - assert "Unknown flag: '7'" == 'Usage: pleas...ative integer'
  - Usage: please input a non-negative integer
  + Unknown flag: '7'
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input large composite] - assert "invalid flag...cedure 'main'" == 'composite'
  - composite
  + invalid flag: 4011
  + /src/prime-number.odin(1:2) undefined entry point procedure 'main'
FAILED test/generated/test_reverse_string.py::test_reverse_string[odin/reverse-string.odin-No input] - assert "/src/reverse...ed: 'reverse'" == ''
  + /src/reverse-string.odin(9:5) Syntax Error: Unknown tag directive used: 'reverse'
FAILED test/generated/test_reverse_string.py::test_reverse_string[odin/reverse-string.odin-Empty input] - assert "/src/reverse...ed: 'reverse'" == ''
  + /src/reverse-string.odin(9:5) Syntax Error: Unknown tag directive used: 'reverse'
FAILED test/generated/test_reverse_string.py::test_reverse_string[odin/reverse-string.odin-Ascii String] - assert "Invalid flag...ed: 'reverse'" == 'dlroW ,olleH'
  - dlroW ,olleH
  + Invalid flag: Hello, World
  + /src/reverse-string.odin(9:5) Syntax Error: Unknown tag directive used: 'reverse'
================== 15 failed, 402 passed in 104.49s (0:01:44) ==================
```

**Comment by jrg94** [comment]
Out of my own curiosity, I added back the commented out parts at the end of the reverse string solution. I'll see how testing goes before adding the other commented out piece back.

**Comment by jrg94** [comment]
Ended up getting some slightly different errors. Something about an unknown tag directive:

```
=========================== short test summary info ============================
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input 0] - assert "invalid flag...cedure 'main'" == 'composite'
  - composite
  + invalid flag: 0
  + /src/prime-number.odin(1:2) undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input 1] - assert "invalid flag...cedure 'main'" == 'composite'
  - composite
  + invalid flag: 1
  + /src/prime-number.odin(1:2) undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input 2] - assert "invalid flag...cedure 'main'" == 'prime'
  - prime
  + invalid flag: 2
  + /src/prime-number.odin(1:2) undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input small composite] - assert "invalid flag...cedure 'main'" == 'composite'
  - composite
  + invalid flag: 4
  + /src/prime-number.odin(1:2) undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input small prime] - assert "invalid flag...cedure 'main'" == 'prime'
  - prime
  + invalid flag: 7
  + /src/prime-number.odin(1:2) undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input large composite] - assert "invalid flag...cedure 'main'" == 'composite'
  - composite
  + invalid flag: 4011
  + /src/prime-number.odin(1:2) undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input large prime] - assert "invalid flag...cedure 'main'" == 'prime'
  - prime
  + invalid flag: 3727
  + /src/prime-number.odin(1:2) undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_invalid[odin/prime-number.odin-invalid input: not a number] - assert "Invalid flag...cedure 'main'" == 'Usage: pleas...ative integer'
  - Usage: please input a non-negative integer
  + Invalid flag: a
  + /src/prime-number.odin(1:2) Undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_invalid[odin/prime-number.odin-no input] - assert "/src/prime-n...cedure 'main'" == 'Usage: pleas...ative integer'
  - Usage: please input a non-negative integer
  + /src/prime-number.odin(1:2) Undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_invalid[odin/prime-number.odin-invalid input: not an integer] - assert "Invalid flag...cedure 'main'" == 'Usage: pleas...ative integer'
  - Usage: please input a non-negative integer
  + Invalid flag: 6.7
  + /src/prime-number.odin(1:2) Undefined entry point procedure 'main'
FAILED test/generated/test_prime_number.py::test_prime_invalid[odin/prime-number.odin-invalid input: negative] - assert "Unknown flag: '7'" == 'Usage: pleas...ative integer'
  - Usage: please input a non-negative integer
  + Unknown flag: '7'
FAILED test/generated/test_prime_number.py::test_prime_invalid[odin/prime-number.odin-empty input] - assert "/src/prime-n...cedure 'main'" == 'Usage: pleas...ative integer'
  - Usage: please input a non-negative integer
  + /src/prime-number.odin(1:2) Undefined entry point procedure 'main'
FAILED test/generated/test_reverse_string.py::test_reverse_string[odin/reverse-string.odin-No input] - assert "/src/reverse...ed: 'reverse'" == ''
  + /src/reverse-string.odin(9:5) Syntax Error: Unknown tag directive used: 'reverse'
FAILED test/generated/test_reverse_string.py::test_reverse_string[odin/reverse-string.odin-Empty input] - assert "/src/reverse...ed: 'reverse'" == ''
  + /src/reverse-string.odin(9:5) Syntax Error: Unknown tag directive used: 'reverse'
FAILED test/generated/test_reverse_string.py::test_reverse_string[odin/reverse-string.odin-Ascii String] - assert "Invalid flag...ed: 'reverse'" == 'dlroW ,olleH'
  - dlroW ,olleH
  + Invalid flag: Hello, World
  + /src/reverse-string.odin(9:5) Syntax Error: Unknown tag directive used: 'reverse'
```

**Comment by jrg94** [comment]
Looking better! Just getting some invalid flag errors now. Not exactly sure what that means. 

```
=========================== short test summary info ============================
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input small prime] - AssertionError: assert 'invalid flag...ative integer' == 'prime'
  - prime
  + invalid flag: 7
  + usage: please input a non-negative integer
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input 0] - AssertionError: assert 'invalid flag...ative integer' == 'composite'
  - composite
  + invalid flag: 0
  + usage: please input a non-negative integer
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input large composite] - AssertionError: assert 'invalid flag...ative integer' == 'composite'
  - composite
  + invalid flag: 4011
  + usage: please input a non-negative integer
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input 1] - AssertionError: assert 'invalid flag...ative integer' == 'composite'
  - composite
  + invalid flag: 1
  + usage: please input a non-negative integer
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input large prime] - AssertionError: assert 'invalid flag...ative integer' == 'prime'
  - prime
  + invalid flag: 3727
  + usage: please input a non-negative integer
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input 2] - AssertionError: assert 'invalid flag...ative integer' == 'prime'
  - prime
  + invalid flag: 2
  + usage: please input a non-negative integer
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input small composite] - AssertionError: assert 'invalid flag...ative integer' == 'composite'
  - composite
  + invalid flag: 4
  + usage: please input a non-negative integer
FAILED test/generated/test_prime_number.py::test_prime_invalid[odin/prime-number.odin-invalid input: not a number] - AssertionError: assert 'Invalid flag...ative integer' == 'Usage: pleas...ative integer'
  + Invalid flag: a
    Usage: please input a non-negative integer
FAILED test/generated/test_prime_number.py::test_prime_invalid[odin/prime-number.odin-invalid input: not an integer] - AssertionError: assert 'Invalid flag...ative integer' == 'Usage: pleas...ative integer'
  + Invalid flag: 6.7
    Usage: please input a non-negative integer
FAILED test/generated/test_prime_number.py::test_prime_invalid[odin/prime-number.odin-invalid input: negative] - assert "Unknown flag: '7'" == 'Usage: pleas...ative integer'
  - Usage: please input a non-negative integer
  + Unknown flag: '7'
FAILED test/generated/test_reverse_string.py::test_reverse_string[odin/reverse-string.odin-No input] - assert "/src/reverse...ed: 'reverse'" == ''
  + /src/reverse-string.odin(9:5) Syntax Error: Unknown tag directive used: 'reverse'
FAILED test/generated/test_reverse_string.py::test_reverse_string[odin/reverse-string.odin-Empty input] - assert "/src/reverse...ed: 'reverse'" == ''
  + /src/reverse-string.odin(9:5) Syntax Error: Unknown tag directive used: 'reverse'
FAILED test/generated/test_reverse_string.py::test_reverse_string[odin/reverse-string.odin-Ascii String] - assert "Invalid flag...ed: 'reverse'" == 'dlroW ,olleH'
  - dlroW ,olleH
  + Invalid flag: Hello, World
  + /src/reverse-string.odin(9:5) Syntax Error: Unknown tag directive used: 'reverse'
================== 13 failed, 404 passed in 99.75s (0:01:39) ===================
```

**Comment by jrg94** [comment]
Only one bug now! The program is treating 2 as composite when it is prime. 

```
=========================== short test summary info ============================
FAILED test/generated/test_prime_number.py::test_prime_valid[odin/prime-number.odin-sample input 2] - AssertionError: assert 'composite' == 'prime'
  - prime
  + composite
================== 1 failed, 413 passed in 103.55s (0:01:43) ===================
```

**Review by jrg94 — APPROVED** [review]
👍🏼

**Pull request merged**